Objects/object.c

```diff
@@ -282,13 +282,10 @@ PyObject_Print(PyObject *op, FILE *fp, int flags)
                 s = PyObject_Str(op);
             else
                 s = PyObject_Repr(op);
-            if (s == NULL)
+            if (s == NULL) {
                 ret = -1;
-            else if (PyBytes_Check(s)) {
-                fwrite(PyBytes_AS_STRING(s), 1,
-                       PyBytes_GET_SIZE(s), fp);
             }
-            else if (PyUnicode_Check(s)) {
+            else {
                 PyObject *t;
```

Comment:
You can add an `assert(PyUnicode_Check(s));` here to validate the behavior.

Reply:
Please do. Assert is only enabled in debug builds, there's no reason to avoid it.